### PR TITLE
Fix --stop-in-app and --stop-in-init options.

### PR DESCRIPTION
**Pull request by awlauria**
Some backend glue was missing.

Signed-off-by: Austen Lauria <awlauria@us.ibm.com>

### CODE REVIEW

File: src/prted/pmix/pmix_server_dyn.c [hunk 1]
```diff
@@ -533,12 +533,14 @@ static void interim(int sd, short args, void *cbdata)
             if (PMIX_PROC_RANK == info->value.type) {
                 prte_set_attribute(&jdata->attributes, PRTE_JOB_STOP_IN_INIT, PRTE_ATTR_GLOBAL,
                                    &info->value.data.rank, PMIX_PROC_RANK);
+                PMIX_INFO_LOAD(info, PMIX_DEBUG_STOP_IN_INIT, &info->value.data.rank, PMIX_PROC_RANK);
```
Comment (rhc54) [inline src/prted/pmix/pmix_server_dyn.c:536]: I'm not sure I understand this - `info` is the incoming object sent to us via `PMIx_Spawn`, so (a) we really don't want to be modifying it, and (b) it already contains the value you are loading here. Can you perhaps explain the reasoning for this change?

File: src/prted/pmix/pmix_server_dyn.c [hunk 2]
```diff
@@ -549,9 +551,11 @@ static void interim(int sd, short args, void *cbdata)
             pmix_server_cache_job_info(jdata, info);
 
         } else if (PMIX_CHECK_KEY(info, PMIX_DEBUG_STOP_IN_APP)) {
+            rank = PMIX_RANK_WILDCARD;
             prte_set_attribute(&jdata->attributes, PRTE_JOB_STOP_IN_APP, PRTE_ATTR_GLOBAL,
-                               info->value.data.string, PMIX_STRING);
```
Comment (rhc54) [inline src/prted/pmix/pmix_server_dyn.c:553]: I'm not sure this is correct - I'll have to look in the code, but we need to have that label somewhere so we know where to tell the app to stop.

File: src/prted/pmix/pmix_server_dyn.c [hunk 3]
```diff
@@ -549,9 +551,11 @@ static void interim(int sd, short args, void *cbdata)
...
+                              &rank, PMIX_PROC_RANK);
             /* also must add to job-level cache */
+            PMIX_INFO_LOAD(info, PMIX_DEBUG_STOP_IN_APP, &rank, PMIX_PROC_RANK);
```
Comment (rhc54) [inline src/prted/pmix/pmix_server_dyn.c:558]: You really don't want to do this - you are overwriting what we are being told and losing the label information that the app requires.

File: src/tools/prte/prte.c [hunk 4]
```diff
@@ -885,6 +885,14 @@ int main(int argc, char *argv[])
         PMIX_INFO_LIST_ADD(ret, jinfo, PMIX_DEBUG_STOP_ON_EXEC, NULL, PMIX_BOOL);
     }
 
+    if (pmix_cmd_line_is_taken(&results, PRTE_CLI_STOP_IN_INIT)) {
```
Comment (rhc54) [inline src/tools/prte/prte.c:888]: This one needs to look at the rank and see if we should be telling it to stop. Probably needs to move down to where we load the per-proc data if we are really going to support per-proc "stop" directives.

**Review by rhc54 — COMMENTED** [review]
Something just doesn't seem right here, but it will require some reviewing of the Standard and our implementation to identify the problem. I'll probably need a little time to resolve it due to a "bad back" day.

**Comment by rhc54** [comment]
Having trouble trying to work on this as OMPI main branch appears to be broken - see https://github.com/open-mpi/ompi/issues/10869

**Comment by awlauria** [comment]
Thanks - https://github.com/open-mpi/ompi/issues/10869 is now merged

**Comment by rhc54** [comment]
I believe this has now been replaced by a combination of OMPI and PRRTE changes.

**Pull request closed without merging**